### DISCUSSION TITLE
Slow response of laser sensor VL53L0X

### DISCUSSION THREAD

**Discussion by tasmotari** [arendst/Tasmota#17188] [Q&A]
Hello there...

I try to use laser distance sensor VL53L0X as trigger (like motion), but response is very slow 2-4 sec (via rules).

Is the only way to setup this sensor as trigger via RULES?
Is there any way to setup interrupts from special GPIO1 pin on sensor to do this on tasmota?

**Comment by barbudor** [comment]
The VL53L0X driver is reading the ensor every 250ms
The rule engine is polling the sensors every 100ms
So I'm suprised that you get that so slowly.

What are your rules ?

**Reply by sfromis** [reply]
If you are judging the response time by looking at the web UI, that does not work. By default the html page is updated at [2.345 second](https://tasmota.github.io/docs/Commands/#webrefresh) intervals.

**Reply by tasmotari** [reply]
Rule1
ON VL53L0X#Distance>10 DO POWER ON ENDON
ON VL53L0X#Distance=null DO POWER OFF ENDON

Rule1 1
Rule1 5

**Reply by tasmotari** [reply]
> If you are judging the response time by looking at the web UI, that does not work. By default the html page is updated at [2.345 second](https://tasmota.github.io/docs/Commands/#webrefresh) intervals.

@sfromis On web interface "VL53L0X Distance" data updates close to 1s, but rule triggered (in my example POWER ON/OFF) with a delay...

**Reply by tasmotari** [reply]
@barbudor I also noticed in rules when "Once" is OFF, delay between executions is 4-5s. ESP32-D0WD-V3 rev.3
